include/boost/graph/link_cut_trees.hpp

```diff
@@ -0,0 +1,277 @@
+//
+//=======================================================================
+// Copyright 2019
+// Author: Yi Ji
+//
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//=======================================================================
+//
+#ifndef BOOST_LINK_CUT_TREES_HPP
+#define BOOST_LINK_CUT_TREES_HPP
+
+#include <boost/graph/properties.hpp>
+#include <boost/unordered_map.hpp>
+
+namespace boost
+{
+    template<class ElementParentMap, class ElementChildMap = ElementParentMap>
```

Comment:
I presume that `Element` in the Parent and Child maps have to be the same type, right? Or at the very minimum, they must be convertible to each other? The reason I ask is that I presume we can simplify the member functions from being templated on `Element` to being non-template functions. If we impose the strict requirement of being the same type then we can just get the element type from one of the map types, something like:
```
using Element = ElementParentMap::key_type;
```
Or if we want to allow more flexibility, then it might have to be:
```
using Element = typename std::common_type_t<ElementParentMap::key_type, ElementChildMap::key_type>;
```

What do you think?